lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -410,27 +464,18 @@ function setupEventListeners(transport) {
       transport.disconnect();
       return;
     }
-    if (transport._reconnectAttemptsLeft <= 0) {
+
+    const reconnectTimer = transport._getReconnectTimer();
+    if (!reconnectTimer) {
```

Comment:
@makarandp0 I think this ensures that if the timeout happens in the middle of a reconnect attempt, then the attempt is allowed to finish. Is that correct?

Reply:
This check ensures that we do not try to reconnect if timer has expired (or if its 1st time connecting)

Timer expiring does not initiate  disconnect, so that ensures it will not affect ongoing attempt.